src/modules/HttpServer.rb

```
@@ -851,6 +852,20 @@ def AutoPackages
     publish :function => :Export, :type => "map ()"
     publish :function => :Summary, :type => "list ()"
     publish :function => :AutoPackages, :type => "map ()"
+
+  private
+
+    def backup_vhost_config
+      files = FileChanges.new_created_files(dynamic_files_to_check)
+      return if files.empty?
+
+      backup_dir = File.join(APACHE_VHOSTS_DIR, "YaSTsave")
+      SCR.Execute(path(".target.bash"), "mkdir #{backup_dir}")
```

does APACHE_VHOSTS_DIR have to exist? if not, then this line silently failed.